### PR TITLE
feat(vue): add vue support

### PR DESCRIPTION
**Pull request by PhentomPT**
# Vue3 support

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [x] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [x] Documentation content changes
- [x] Other (please describe): framework support

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### Assets
- [x] Vue logo

### Packages
- [x] vue (wrapper)
- [x] vue-test-app
- [x] docs
   - [x] vue stackblitz runtime 

### Docs

- [x] tree.md
- [x] validation.md
- [x] drawer.md
- [x] messagebar.md
- [x] tile.md
- [x] blind.md
- [x] dropdown.md
- [x] modal.md
- [x] time-picker.md
- [x] breadcrumb.md
- [x] event-list.md
- [x] basic-navigation.md
- [x] popover-news.md
- [x] vertical-tabs.md
- [x] about-and-legal.md
- [x] map-navigation.md
- [x] settings.md
- [x] toast.md
- [x] button.md
- [x] expanding-search.md
- [x] pill.md
- [x] toggle.md
- [x] category-filter.md 
- [x] flip.md
- [x] adiobutton.md
- [x] select.md
- [x] upload.md
- [x] checkbox.md
- [x] group.md
- [x] spinner.md
- [x] chip.md
- [x] html-grid.md
- [x] split-button.md
- [x] workflow.md
- [x] date-picker.md
- [x] input.md
- [x] tabs.md
- [x] date-time-picker.md
- [x] kpi.md
- [x] textarea.md

### Preview examples

- [x] tree.vue
- [x] validations.vue
- [x] modal.vue
- [x] about-and-legal.vueinput-readonly.vue
- [x] aggrid.vueinput-with-icon.vue
- [x] basic-navigation-without-header.vue
- [x] input.vue
- [x] basic-navigation.vue
- [x] kpi.vue
- [x] blind.vue
- [x] map-navigation-overlay.vue
- [x] breadcrumb-next-items.vue
- [x] map-navigation.vue
- [x] breadcrumb-truncate.vue
- [x] menu-with-bottom-tabs.vue
- [x] breadcrumb.vue
- [x] message-bar.vue
- [x] button-ghost.vue
- [x] button-grey-ghost.vue
- [x] pill.vue
- [x] button-grey-secondary.vue
- [x] popover-news.vue
- [x] button-grey.vue
- [x] radio-button.vue
- [x] button-group.vue
- [x] select-editable.vue
- [x] button-secondary.vue
- [x] select-multiple.vue
- [x] button-selected.vue
- [x] select.vue
- [x] button-text-icon.vue
- [x] settings.vue
- [x] button-with-icon.vue
- [x] spinner-large.vue
- [x] buttons.vue
- [x] spinner.vue
- [x] category-filter-suggestions.vue
- [x] split-button-icons.vue
- [x] category-filter.vue
- [x] split-button.vue
- [x] checkbox-indeterminate.vue
- [x] tabs-rounded.vue
- [x] checkbox.vue
- [x] tabs.vue
- [x] chip.vue
- [x] textarea-disabled.vue
- [x] datepicker-range.vue
- [x] textarea-readonly.vue
- [x] datepicker.vue
- [x] textarea.vue
- [x] datetimepicker.vue
- [x] tile.vue
- [x] drawer-full-height.vue
- [x] timepicker.vue
- [x] drawer.vue
- [x] toast-custom-element.vue
- [x] dropdown-icon.vuetoast-custom.vue
- [x] dropdown.vue
- [x] toast.vue
- [x] event-list-compact.vue
- [x] toggle-color.vue
- [x] event-list-custom-item-height.vue
- [x] toggle-custom-label.vue
- [x] event-list-selected.vue
- [x] toggle-disabled.vue
- [x] event-list.vue
- [x] toggle.vue
- [x] expanding-search.vue
- [x] flip-tile.vue
- [x] upload.vue
- [x] group-context-menu.vue
- [x] vertical-tabs-with-avatar.vue
- [x] group-custom-entry.vue
- [x] vertical-tabs.vue
- [x] group-header-suppressed.vue
- [x] workflow-vertical.vue
- [x] group.vue
- [x] workflow.vue
- [x] input-disabled.vue

### CODE REVIEW

**Comment by danielleroux** [comment]
@PhentomPT in one of your latest JF we decided to ignore vue for now (https://github.com/siemens/ix/issues/105#issuecomment-1334815157).

**Comment by PhentomPT** [comment]
> @PhentomPT in one of your latest JF we decided to ignore vue for now ([#105 (comment)](https://github.com/siemens/ix/issues/105#issuecomment-1334815157)). Is there any new reason to implement the wrapper?

@danielleroux yes, here is a list on some important reasons

- future projects in vue that want to implement the library
- major framework with a big community
- stencil wrapper is available, very simple & boosts integration vs pain javascript usage
- high compatibility & low effort for documentation

**Comment by danielleroux** [comment]
> > @PhentomPT in one of your latest JF we decided to ignore vue for now ([#105 (comment)](https://github.com/siemens/ix/issues/105#issuecomment-1334815157)). Is there any new reason to implement the wrapper?
> 
> @danielleroux yes, here is a list on some important reasons
> 
> * future projects in vue that want to implement the library
> * major framework with a big community
> * stencil wrapper is available, very simple & boosts integration vs pain javascript usage
> * high compatibility & low effort for documentation

We have to clarify this in the next jourfix

File: packages/documentation/docs/installation/vue.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+sidebar_position: 2
+sidebar_title: Vue
+title: Vue
+---
+
```
Comment (danielleroux) [inline packages/documentation/docs/installation/vue.md:6]: We should also add here a disclaimer that what `experimental` mean for us

File: packages/documentation/docs/installation/vue.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+### Usage
+
+```ts
+import '@siemens/ix-icons/dist/css/ix-icons.css';
+import '@siemens/ix/dist/siemens-ix/siemens-ix.css';
+
+import { createApp } from 'vue';
+import { ixPlugin } from '@siemens/ix-vue/dist';
```
Comment (danielleroux) [inline packages/documentation/docs/installation/vue.md:30]: We should avoid the additional path `dist`

File: packages/documentation/static/code-runtime/vue/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "vue": "^3.2.6",
+    "@siemens/ix": "^1",
+    "@siemens/ix-icons": "^1",
+    "@siemens/ix-vue": "^1"
+  },
+  "devDependencies": {
+    "@vitejs/plugin-vue": "^1.6.1",
+    "@vue/compiler-sfc": "^3.2.6",
+    "typescript": "^4.3.2",
+    "vite": "^2.5.4",
```
Comment (danielleroux) [inline packages/documentation/static/code-runtime/vue/package.json:20]: We should at least align the major version of vite with the other versions inside the repo. `vite with major version 3`

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/documentation/static/code-runtime/vue/package.json (hunk 3 above)
Comment (PhentomPT) [reply packages/documentation/static/code-runtime/vue/package.json:20]: there is already version 4, we could go straight to that one and also update the remaining repo dependencies

File: packages/documentation/docs/installation/vue.md (hunk 1 above)
Comment (PhentomPT) [reply packages/documentation/docs/installation/vue.md:6]: what would that explanation be?

File: packages/documentation/static/code-runtime/vue/package.json (hunk 3 above)
Comment (danielleroux) [reply packages/documentation/static/code-runtime/vue/package.json:20]: Also great you can do this inside this PR or in a separate one.

File: packages/documentation/docs/installation/vue.md (hunk 1 above)
Comment (danielleroux) [reply packages/documentation/docs/installation/vue.md:6]: @nuke-ellington any suggestions how we should describe the experimental state of an wrapper component?

What we will not do for experimental wrappers are:

- Blueprints
- Support
- Bugfixes after react/angular
- ...

**Comment by PhentomPT** [comment]
@danielleroux MR is ready

**Comment by danielleroux** [comment]
> @danielleroux MR is ready

This is one of our main goals for 1.5.0 but as discussed in last team jf we have first to make the update to stencil v3.0.0 (https://github.com/siemens/ix/issues/327)

**Comment by danielleroux** [comment]
I resolve merge conflicts after https://github.com/siemens/ix/pull/431 is merged to main

**Review by danielleroux — APPROVED** [review]
lgtm. Maybe we have to adapt the installation disclaimer after merge.

**Pull request merged**